apps/devhub/views.py

```diff
@@ -1393,7 +1393,13 @@ def submit(request, step):
         response.set_cookie(DEV_AGREEMENT_COOKIE)
         return response
 
-    return render(request, 'devhub/addons/submit/start.html', {'step': step})
+    if request.user.read_dev_agreement is None:
+        return render(request, 'devhub/addons/submit/start.html',
+                      {'step': step})
+    else:
+        response = redirect(_step_url(2))
+        response.set_cookie(DEV_AGREEMENT_COOKIE)
```

Comment:
this seems like the wrong place to set a cookie. In fact, I can't see any reason to use a cookie at all and I think you should remove it with this patch. In [submit_addon()](https://github.com/Nolski/olympia/blob/signing-api-agreement/apps/devhub/views.py#L1408) you can change `if DEV_AGREEMENT_COOKIE not in request.COOKIES` to `if not request.user.read_dev_agreement`. After that you can rename some of the cookie tests -- they will still be valid but the implementation will have changed.